doc/man7/provider-cipher.pod

```diff
@@ -29,6 +29,12 @@ provider-cipher - The cipher library E<lt>-E<gt> provider functions
  int OSSL_FUNC_cipher_decrypt_init(void *cctx, const unsigned char *key,
                                    size_t keylen, const unsigned char *iv,
                                    size_t ivlen, const OSSL_PARAM params[]);
+ int OSSL_FUNC_cipher_encrypt_skey_init(void *cctx, void *skeydata,
+                                        const unsigned char *iv, size_t ivlen,
+                                        const OSSL_PARAM params[]);
+ int OSSL_FUNC_cipher_encrypt_skey_init(void *cctx, void *skeydata,
+                                        const unsigned char *iv, size_t ivlen,
+                                        const OSSL_PARAM params[]);
```

Comment:
There's one thing missing here...

If we fetch a cipher like ASE-128-CBC, how to figure out the corresponding skeymgmt algorithm?

With keypairs, this was recently solved by allowing the signature implementations to specify what key type(s) are compatible with them:

https://github.com/openssl/openssl/blob/d69c014608acdfa37839d49412e6d6974ac539a0/doc/man7/provider-signature.pod#L25-L26

Reply:
This is a high impact issue, sorry that I didn't notice it before

Reply:
I'd ask you to continue the discussion in #26702 if necessary.

The current model is the same:
When we create a key, we fetch the SKEYMGMT by name. The provider can allow any set of aliases for the particular SKEYMGMT: they can have separate keymgmt for AES-128-CBC and AES-128-CTR or they can implement one generic with the only length check.

When we set the key, we double-check that the provider for the SKEYMGMT is equal to the provider for the CIPHER. It means that we can be sure that the provider knows about the internal structure. Could you please elaborate which extra checks you think are necessary?

I'll double-check the usage of the `OSSL_FUNC_signature_query_key_types` function

Reply:
If you alias AES-128-CBC and AES-128-CTR as the same, you're getting yourself into a pile of trouble.  For example, someone wanting to check which cipher they are currently using with something like `EVP_CIPHER_is_a(cipher, "AES-192-CBC")` will find that matching *any* algorithm that you have aliased this way.

This is a bad idea

Reply:
@levitte i proposed to Dmitry a while ago to provide a mapping table (cipher -> key type) that would be used by skeymgmt, it would have to be something the providers return, because we can't know in advance all the cipher and what kind of key they may accept in the future.
It would look similar to TLS groups, which are queried from providers.

Reply:
Fix landed in #26702